include/macroni/Analysis/Kernel/RCUAnalysis.hpp

```diff
@@ -13,12 +15,48 @@ struct rcu_analysis {
   virtual ~rcu_analysis() = default;
 
 protected:
-  void warn_rcu_dereference(RCU_Dereference_Interface &op);
   void analyze_non_rcu_function(vast::hl::FuncOp &func);
   auto walk_until_call(llvm::StringRef name);
   void analyze_acquires_function(vast::hl::FuncOp &func);
   void analyze_releases_function(vast::hl::FuncOp &func);
   void analyze_critical_section(RCUCriticalSection cs);
-  virtual void emit(std::string warning) { llvm::errs() << warning; }
+
+  // Format an operation's location as a string for diagnostics.
+  virtual std::string format_location(mlir::Operation *op) {
+    std::string s;
+    auto os = llvm::raw_string_ostream(s);
+    op->getLoc()->print(os);
+    s.erase(s.find("loc("), 4);
+    s.erase(s.find('"'), 1);
+    s.erase(s.find('"'), 1);
+    s.erase(s.rfind(')'), 1);
```

Comment:
This can be achieved by specifying `mlir::OpPrintingFlags` class that is passed to `print`.

Reply:
Ooh did not realize will do